### PR TITLE
adds GET `/auctions` integration tests

### PR DESCRIPTION
**Pull request by swarajpure**

### CODE REVIEW

File: test/integration/auction.test.js [hunk 1]
```diff
@@ -47,13 +50,33 @@ describe('Auctions', function () {
...
           expect(res.body.auctions).to.be.a('array')
+
+          return done()
+        })
+    })
+
+    it('Should return 404, for Auction not Found', function (done) {
+      chai
+        .request(app)
+        .get('/auction')
```
Comment (swarajpure) [inline test/integration/auction.test.js:61]: wait, this route itself seems wrong 😅

File: test/integration/auction.test.js [hunk 2]
```diff
@@ -47,13 +50,33 @@ describe('Auctions', function () {
...
+          if (err) { return done(err) }
+
+          expect(res).to.have.status(404)
+          expect(res.body).to.be.a('object')
+          expect(res.body).to.deep.equal({
+            statusCode: 404,
+            error: 'Not Found',
+            message: 'Not Found'
+          })
+
```
Comment (swarajpure) [inline test/integration/auction.test.js:72]: What is this test case checking? 🤔

File: test/integration/auction.test.js [hunk 3]
```diff
@@ -62,7 +85,152 @@ describe('Auctions', function () {
...
+          return done()
+        })
+    })
+  })
```
Comment (swarajpure) [inline test/integration/auction.test.js:163]: Again, what is this doing? 🤔

File: test/integration/auction.test.js [hunk 4]
```diff
@@ -62,7 +85,152 @@ describe('Auctions', function () {
...
+        .end((err, res) => {
+          if (err) { return done(err) }
+
+          expect(res).to.have.status(403)
+          expect(res.body).to.be.a('object')
+          expect(res.body.message).to.be.equal('Your bid was not higher than current one!')
+
+          return done()
+        })
+    })
```
Comment (swarajpure) [inline test/integration/auction.test.js:198]: Can we also have a test for the case when the the user doesn't have sufficient balance?

**Review by swarajpure — COMMENTED** [review]
Having some queries, please check

File: test/integration/auction.test.js (hunk 1 above)
Comment (PremanandChowdhury) [reply test/integration/auction.test.js:61]: In thought of creating the 404 error i came up with this condition.
I'll make the appropriate changes.

File: test/integration/auction.test.js (hunk 2 above)
Comment (PremanandChowdhury) [reply test/integration/auction.test.js:72]: Same as above 😓

File: test/integration/auction.test.js (hunk 3 above)
Comment (PremanandChowdhury) [reply test/integration/auction.test.js:163]: Same

File: test/integration/auction.test.js (hunk 4 above)
Comment (PremanandChowdhury) [reply test/integration/auction.test.js:198]: OK, Will be done.

Comment (PremanandChowdhury) [reply test/integration/auction.test.js:198]: What should be the expected minimum balance ?

File: test/integration/auction.test.js (hunk 3 above)
Comment (PremanandChowdhury) [reply test/integration/auction.test.js:163]: For this the `statusCode : 401` and it is changed in the local.
Route: `/auctions`, for Unathorized user

File: test/integration/auction.test.js [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+            message: 'Unauthenticated User'
+          })
+
+          return done()
+        })
+    })
+  })
+
+  describe('POST /auctions/bid/:id', function () {
+    it('Should make a new bid with given auctionId', function (done) {
```
Comment (swarajpure) [inline test/integration/auction.test.js:147]: NIT: `Should place a new bid.....`

File: test/integration/auction.test.js [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+
+          expect(res).to.have.status(201)
+          expect(res.body).to.be.a('object')
+          expect(res.body.message).to.be.equal('Successfully placed bid!')
+
+          return done()
+        })
+    })
+
+    it('Should have a sufficient balance', function (done) {
```
Comment (swarajpure) [inline test/integration/auction.test.js:164]: `User should have sufficient balance for bidding` would make more sense to me personally. What do you think?

File: test/integration/auction.test.js [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+
+          expect(res).to.have.status(403)
+          expect(res.body).to.be.a('object')
+          expect(res.body.message).to.be.equal('You do not have sufficient money')
+
+          return done()
+        })
+    })
+
+    it('Should be higher than the previous bid', function (done) {
```
Comment (swarajpure) [inline test/integration/auction.test.js:182]: `Bid` should be higher than the previous bid

File: test/integration/auction.test.js [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+
+          expect(res).to.have.status(201)
+          expect(res.body).to.be.a('object')
+          expect(res.body.message).to.be.equal('Auction created successfully!')
+
+          return done()
+        })
+    })
+
+    it('Should have enough Item Type in wallet', function (done) {
```
Comment (swarajpure) [inline test/integration/auction.test.js:109]: `User should have enough items in wallet to sell` What do you think?

**Review by swarajpure — COMMENTED** [review]
Just some small changes in the tests titles/messages
Rest everything looks perfect!

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]

File: test/integration/auction.test.js [hunk 9]
```diff
@@ -0,0 +1,218 @@
...
+        .request(app)
+        .get('/auctions')
+        .end((err, res) => {
+          if (err) { return done(err) }
+
+          expect(res).to.have.status(200)
+          expect(res.body).to.be.a('object')
+          expect(res.body).to.have.all.keys(...auctionKeys)
+          expect(res.body.message).to.be.equal('Auctions returned successfully!')
+          expect(res.body.auctions).to.be.a('array')
```
[the remark this answers is not in the document]
Comment (PremanandChowdhury) [reply test/integration/auction.test.js:51]: Can you please explain little more detailed, please 😅
I am new to the tests.

[the remark this answers is not in the document]
Comment (PremanandChowdhury) [reply test/integration/auction.test.js:51]: This is what I am thinking : `expect(res.body.auctions).to.be.a('array').to.have.all.keys(...auctionKeys)` 
Is it what you are saying ?

[the remark this answers is not in the document]
Comment (ankurnarkhede) [reply test/integration/auction.test.js:51]: I mean, also assert on the data type and value of the response keys for the value of the keys in any one test case.

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]

**Comment by PremanandChowdhury** [comment]
I made the changes please review, before 1May it will be helpful. @ankurnarkhede [Link](https://github.com/Real-Dev-Squad/website-backend/pull/310/commits/a9145c56787eb79364f3985ecd98f1b9ae65ba73)

The `GET /tasks` is failing the tests here. Also, I checked in the local it is not failing, it is working fine.

For the `/auctions` Unit Test, can I make the PR after the break.

**Review by ankurnarkhede — APPROVED** [review]

**Pull request merged**